src/libs/actions/App.ts

```diff
@@ -398,7 +398,7 @@ function setUpPoliciesAndNavigate(session: OnyxEntry<OnyxTypes.Session>) {
         return;
     }
     if (!isLoggingInAsNewUser && exitTo) {
-        Navigation.isNavigationReady()
+        Navigation.waitForProtectedRoutes()
```

Comment:
what does this do?

Reply:
In addition to waiting for the navigation to be "ready", `waitForProtectedRoutes` waits for one of our authenticated routes (like the LHN or a report) to be rendered. 

Atm `isNavigationReady` is a bit of a hack (it gets triggered [here](https://github.com/Expensify/App/blob/744e35c85157ecef9cdfab81a0ab56891dc507f9/src/libs/Navigation/NavigationRoot.tsx#L38)) and doesn't work very well in testing , but this works consistently.